client/login/magic-login/index.jsx

```diff
@@ -75,6 +75,12 @@ class MagicLogin extends React.Component {
 		if ( emailAddress ) {
 			loginParameters.emailAddress = emailAddress;
 		}
+
+		const signupUrl = get( this.props, [ 'query', 'signup_url' ] );
+		if ( signupUrl ) {
+			loginParameters.signupUrl = signupUrl;
+		}
```

Comment:
This could be shortened to
```js
loginParameters.signupUrl = this.props.query?.signup_url;
```
The `login()` function can handle null-ish `signupUrl` parameter.

Reply:
Addressed in https://github.com/Automattic/wp-calypso/pull/54646/commits/032fd7d1c84cd74010825d8fa72b504c0b7d98c4